libs/sdk-core/src/ldk/node_api.rs

```diff
@@ -127,92 +156,82 @@ impl NodeAPI for Ldk {
         _max_hops: u32,
         _last_hop: Option<&'a RouteHintHop>,
     ) -> NodeResult<Vec<MaxChannelAmount>> {
-        Err(NodeError::Generic(
-            "LDK implementation not yet available".to_string(),
-        ))
+        Err(NodeError::generic("LDK implementation not yet available"))
     }
 
     async fn redeem_onchain_funds(
         &self,
         _to_address: String,
         _sat_per_vbyte: u32,
     ) -> NodeResult<Vec<u8>> {
-        Err(NodeError::Generic(
-            "LDK implementation not yet available".to_string(),
-        ))
+        Err(NodeError::generic("LDK implementation not yet available"))
     }
 
     async fn prepare_redeem_onchain_funds(
         &self,
         _req: PrepareRedeemOnchainFundsRequest,
     ) -> NodeResult<PrepareRedeemOnchainFundsResponse> {
-        Err(NodeError::Generic(
-            "LDK implementation not yet available".to_string(),
-        ))
+        Err(NodeError::generic("LDK implementation not yet available"))
     }
 
-    async fn start_signer(&self, _shutdown: mpsc::Receiver<()>) {
-        // No-op for stub implementation
+    async fn start_signer(&self, mut shutdown: mpsc::Receiver<()>) {
+        debug!("Starting LDK Node");
+        if let Err(e) = self.node.start() {
+            error!("Failed to start LDK Node: {e}");
+            return;
+        }
+        debug!("LDK Node started");
+
+        _ = shutdown.recv().await;
+        debug!("Stopping LDK Node");
+        if let Err(e) = self.node.stop() {
+            error!("Error on stopping LDK Node: {e}");
+        }
+        debug!("LDK Node stopped");
```

Comment:
No external signer in ldk, does it sounds OK to wire the node start here?
Perhaps we use the start/stop from the breez services?
Or as an alternative perhaps just rename this trait function to "start" instead of "start_signer" which allows each node implementation adapt it to its own startup needs?